homeassistant/components/lutron/fan.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+from typing import Any
+import logging
+
+from homeassistant.components.fan import FanEntity, FanEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from . import LUTRON_CONTROLLER, LUTRON_DEVICES, LutronDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+def setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the Lutron fans."""
+    devs = []
+    for area_name, device in hass.data[LUTRON_DEVICES]["fan"]:
+        dev = LutronFan(
+            area_name, 
+            device, 
+            hass.data[LUTRON_CONTROLLER],
+            device.uuid, 
+            device.name, 
+            None)
+        devs.append(dev)
+
+    add_entities(devs, True)
+
+
+class LutronFan(LutronDevice, FanEntity):
+    _attr_should_poll = False
+
+    def __init__(
+        self, area_name, lutron_device, controller,
+        unique_id: str,
+        name: str,
+        preset_modes: list[str] | None,
+    ) -> None:
+        super().__init__(area_name, lutron_device, controller)
+        self._attr_unique_id = unique_id
+        self._attr_supported_features = FanEntityFeature.SET_SPEED
+        self._attr_speed_count = 4
+        self._prev_percentage: int | None = None
+        self._percentage: int | None = None
+        self._preset_modes = preset_modes
+        self._preset_mode: str | None = None
+        self._oscillating: bool | None = None
+        self._direction: str | None = None
+        self._attr_name = name
```

Comment:
```suggestion
```
`LutronDevice` is already implementing name so maybe we should use that instead?

Reply:
This is utilizing the name from LutronDevice but without implementing it the core makes the name generic.  So this is needed for more consistent naming.